src/openrct2/entity/Guest.cpp

```diff
@@ -5444,6 +5446,44 @@ void Guest::UpdateWalking()
     }
 }
 
+void Guest::UpdateWaitingAtCrossing()
+{
+    if (!IsActionInterruptable())
+    {
+        UpdateAction();
+        Invalidate();
+        if (!IsActionWalking())
+            return;
+    }
+
+    Action = PeepActionType::Idle;
+    NextActionSpriteType = PeepActionSpriteType::WatchRide;
+    SwitchNextActionSpriteType();
+
+    if (HasFoodOrDrink())
+    {
+        if ((scenario_rand() & 0xFFFF) <= 1310)
+        {
+            Action = PeepActionType::EatFood;
+            ActionFrame = 0;
+            ActionSpriteImageOffset = 0;
+        }
+
+        UpdateCurrentActionSpriteType();
+
+        return;
+    }
+
+    if ((scenario_rand() & 0xFFFF) <= 64)
```

Comment:
This only has a 65/65535 (0.1%) chance of happening. This seems extremely low, especially compared to the drinking animation.

Reply:
I guess this is ran every frame for guests that are not holding any food or drinks, so it is more likely to happen then.

Reply:
It does seem like a very minor chance. But as you said, it's ran every frame which means it will trigger occasionally. Especially since it's ran for every guest at every railway crossing. In a test park which isn't even very busy, it gets triggered often enough (I've tested this).

I mostly wanted to prevent batches of guests waving continuously at the crossings, which can occur quite easily with a higher chance.